### PR TITLE
Added quest leave route and initial tests

### PR DESCRIPTION
**Pull request by TheHollidayInn**
This the the quest leave route and initial tests for #6509.

### CODE REVIEW

File: website/src/controllers/api-v3/quests.js [hunk 1]
```diff
@@ -65,4 +65,55 @@ api.inviteToQuest = {
...
+    group.quest.members[user._id] = false;
+    group.markModified('quest.members');
+
+    user.party.quest = Group.cleanQuestProgress();
+    user.markModified('party.quest');
+
+    await group.save();
+    await user.save();
+
+    res.respond(200, {});
```
Comment (crookedneighbor) [inline website/src/controllers/api-v3/quests.js]: This should return the group quest object.

Comment (TheHollidayInn) [inline website/src/controllers/api-v3/quests.js]: Sounds good. I figured that might be the case.

**Comment by TheHollidayInn** [comment]
Updated!

**Comment by paglias** [comment]
Updated version at #6695

**Pull request closed without merging**